dotnet/src/Connectors/Connectors.Memory.Redis/RedisMemoryStore.cs

```diff
@@ -0,0 +1,269 @@
+    {
+        foreach (var key in keys)
+        {
+            var result = await this.InternalGetAsync(collectionName, key, withEmbeddings, cancellationToken).ConfigureAwait(false);
+            if (result != null)
+            {
+                yield return result;
+            }
+            else
+            {
+                yield break;
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<string> UpsertAsync(string collectionName, MemoryRecord record, CancellationToken cancellationToken = default)
+    {
+        record.Key = record.Metadata.Id;
+
+        await this._database.HashSetAsync(GetRedisKey(collectionName, record.Key), new[] {
+            new HashEntry("key", record.Key),
+            new HashEntry("metadata", record.GetSerializedMetadata()),
+            new HashEntry("embedding", MemoryMarshal.Cast<float, byte>(record.Embedding.AsReadOnlySpan()).ToArray()),
+            new HashEntry("timestamp", ToTimestampLong(record.Timestamp))
+        }, flags: CommandFlags.None).ConfigureAwait(false);
+
+        return record.Key;
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<string> UpsertBatchAsync(string collectionName, IEnumerable<MemoryRecord> records, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        foreach (var record in records)
+        {
+            yield return await this.UpsertAsync(collectionName, record, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task RemoveAsync(string collectionName, string key, CancellationToken cancellationToken = default)
+    {
+        await this._database.KeyDeleteAsync(GetRedisKey(collectionName, key), flags: CommandFlags.None).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async Task RemoveBatchAsync(string collectionName, IEnumerable<string> keys, CancellationToken cancellationToken = default)
+    {
+        await this._database.KeyDeleteAsync(keys.Select(key => GetRedisKey(collectionName, key)).ToArray(), flags: CommandFlags.None).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetNearestMatchesAsync(
+        string collectionName,
+        Embedding<float> embedding,
+        int limit,
+        double minRelevanceScore = 0,
+        bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (limit <= 0)
+        {
+            yield break;
+        }
+
+        var query = new Query($"*=>[KNN {limit} @embedding $embedding AS vector_score]")
+                    .AddParam("embedding", MemoryMarshal.Cast<float, byte>(embedding.AsReadOnlySpan()).ToArray())
+                    .SetSortBy("vector_score")
+                    .ReturnFields("key", "metadata", "embedding", "timestamp", "vector_score")
+                    .Limit(0, limit)
+                    .Dialect(2);
```

Comment:
What is `2` in this case? Can we put it in separate const variable with naming? It would be also nice to leave a link to documentation about this property.

Reply:
Done.